src/main/java/io/confluent/connect/hdfs/avro/AvroRecordWriterProvider.java

```diff
@@ -84,6 +86,14 @@ public void write(SinkRecord record) {
           }
         } catch (IOException e) {
           throw new DataException(e);
+        } catch (AvroRuntimeException ex) {
+          // Making schema null to invalidate file corrupted due to connection failure
+          schema = null;
+          /*
+          Wrapped into ConnectException so that setRetryTimeout() is called in the
+          TopicPartitionWriter class to retry the records.
+          */
+          throw new ConnectException(ex);
```

Comment:
Are we retrying on ConnectException? Ideally we shouldn't retry on ConnectException.

Reply:
There is already retry logic placed on ConnectException in the TopicPartitionWriter class, we have thrown the connect exception to reuse the existing code implementation.